src/tnlp.cc

```diff
@@ -45,7 +45,7 @@ namespace roboptim
       n = static_cast<Index> (solver_.problem ().function ().inputSize ());
       m = static_cast<Index> (constraintsOutputSize ());
 
-      function_t::vector_t x (n);
+      GenericFunction<EigenMatrixSparse>::vector_t x (n);
```

Comment:
You should have typedefs for each supported function types (plus Function), and all the casts should be done once and for all in the initialization and stored. In the end, you should not have to use any of the templated generic type (unless this is just used once in the implementation or something like that).